contract/lib/dns.ts

```diff
@@ -0,0 +1,155 @@
+import { BufferWriter, DNSRecord } from 'dns-js'
+
+const dnsName = (name) => {
+  // strip leading and trailing .
+  const n = name.replace(/^\.|\.$/gm, '')
+
+  const bufLen = n === '' ? 1 : n.length + 2
+  const buf = Buffer.allocUnsafe(bufLen)
+
+  let offset = 0
+  if (n.length) {
+    const list = n.split('.')
+    for (let i = 0; i < list.length; i++) {
+      const len = buf.write(list[i], offset + 1)
+      buf[offset] = len
+      offset += len + 1
+    }
+  }
+  buf[offset++] = 0
+  return (
```

Comment:
Where is the spec of this format defined? Why each byte has a string `0` padding next to its hex representation?

Reply:
I see, it's equivalent to just toString('hex') on the buffer